kong/plugins/oidc/handler.lua

```diff
@@ -93,22 +95,23 @@ function make_oidc(oidcConfig, oidcSessionConfig)
     end
   end
 
-  -- grab X-Requested-With Header to see if request was from browser/ajax
-  local unauth_action = nil
   local ngx_headers = ngx.req.get_headers()
-  if ngx_headers then
-    local xhr_value = ngx_headers["X-Requested-With"]
-    -- was the request ajax/async?
-    if xhr_value == "XMLHttpRequest" then
-      -- reference: https://github.com/zmartzone/lua-resty-openidc/blob/master/lib/resty/openidc.lua#L1436
-      -- set to deny so resty.openidc returns instead of redirects (ends request)
-      ngx.log(ngx.DEBUG, "OidcHandler ajax/async request detected, setting unauth_action = deny")
-      unauth_action = "deny"
-    end
+  local unauth_action = oidcConfig.force_authentication_path and "pass" or nil
```

Comment:
it seems as though the initialized value of this is being set here.  shouldn't the condition here be alongside the conditions below?  to keep all the logic that sets this value in one place

Reply:
seems like this is where the actual magic for this feature occurs.  it may be worth having additional comments.  why are xhr requests denied?  why are non-xhr requests passed?  why is it nil when the force auth path is found?

Reply:
Updated the logic. Take a look and see if it matches more with what you'd expect.